browser/playlist/playlist_service_factory.cc

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/playlist/playlist_service_factory.h"
+
+#include <memory>
+
+#include "base/feature_list.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/components/playlist/features.h"
+#include "brave/components/playlist/playlist_download_request_manager.h"
+#include "brave/components/playlist/playlist_service.h"
+#include "brave/components/playlist/playlist_youtubedown_component_manager.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/common/chrome_isolated_world_ids.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+
+namespace playlist {
+
+// static
+PlaylistServiceFactory* PlaylistServiceFactory::GetInstance() {
+  return base::Singleton<PlaylistServiceFactory>::get();
+}
+
+// static
+PlaylistService* PlaylistServiceFactory::GetForBrowserContext(
+    content::BrowserContext* context) {
+  if (IsPlaylistEnabled(context)) {
+    return static_cast<PlaylistService*>(
+        GetInstance()->GetServiceForBrowserContext(context, true));
+  }
+
+  return nullptr;
+}
+
+// static
+bool PlaylistServiceFactory::IsPlaylistEnabled(
+    content::BrowserContext* context) {
+  return base::FeatureList::IsEnabled(playlist::features::kPlaylist) &&
+         brave::IsRegularProfile(context);
+}
+
+PlaylistServiceFactory::PlaylistServiceFactory()
+    : BrowserContextKeyedServiceFactory(
+          "PlaylistService",
+          BrowserContextDependencyManager::GetInstance()) {
+  PlaylistDownloadRequestManager::SetPlaylistJavaScriptWorldId(
+      ISOLATED_WORLD_ID_CHROME_INTERNAL);
+  playlist_youtubedown_component_manager_.reset(
```

Comment:
why is this owned by the PlaylistServiceFactory? It's not a bad idea vs having yet another "global" in browser process, but wondering if this is a convention used anywhere else?